graph/nullable.md

```diff
@@ -0,0 +1,312 @@
+# Nullable Properties
+
+A nullable property means *only* that the property may have `null` as a value; the "nullability" of a property does not say anything about how a value is set into a property.
+For example, a non-nullable property is *not* required to create a new instance of an entity.
+It only means that the property will have a value when it is retrieved.
+In the case that no value is provided when the entity is created, this means that the service will create one; this value can be specified with the `DefaultValue` attribute, but if the value is contextual and determine at request time, then the property can both be non-nullable *and* have no `DefaultValue` specified. 
+Below are some examples of nullable and non-nullable properties. 
+
+## CSDL
+
+```xml
+<EntitySet Name="servicePrincipals" Type="self.servicePrincipal" />
+...
+<EntityType Name="servicePrincipal">
+  <Key>
+    <PropertyRef Name="id" />
+  </Key>
+  <Property Name="id" Type="Edm.String" Nullable="false" />
+
+  <Property Name="appId" Type="Edm.String" Nullable="false" /> <!--required for creation-->
+  <Property Name="displayName" Type="Edm.String" Nullable="false" />
+  <Property Name="foo" Type="Edm.String" Nullable="true" DefaultValue="testval" />
+  <Property Name="bar" Type="Edm.String" Nullable="false" DefaultValue="differentvalue" />
+  ...
+</EntityType>
+```
+
+## HTTP Requests
+
+### {1} Create a servicePrincipal with no properties
+
+```HTTP
+POST /servicePrincipals
+
+400 Bad Request
+{
+  "error": {
+    "code": "BadRequest",
+    "message": "The 'appId' property is required to create a servicePrincipal."
+  }
+}
+```
+
+### {2} Create a servicePrincipal without a display name
+
+```HTTP
+POST /servicePrincipals
+{
+  "appId": "00000000-0000-0000-0000-000000000001"
+}
+
+201 Created
+{
+  "appId": "00000000-0000-0000-0000-000000000001",
+  "displayName": "some application name",
+  "foo": "testval",
+  "bar": "differentvalue",
+  ...
+}
+```
+Notes:
+1. `displayName` was given a value by the service even though no value was provided by the client
+2. `foo` has the default value as specified by its `DefaultValue` attribute in the CSDL
+3. `bar` has the default value as specified by its `DefaultValue` attribute in the CSDL
+
+### {3} Update the display name of a service principal to null
+
+```HTTP
+PATCH /servicePrincipals/00000000-0000-0000-0000-000000000001
+{
+  "displayName": null
+}
+
+400 Bad Request
+{
+  "error": {
+    "code": "BadRequest",
```

Comment:
NIT: "BadRequest" should be in camelCase like "badRequest"

Reply:
```suggestion
    "code": "badRequest",
```